src/Symfony/Component/DependencyInjection/Tests/Compiler/AutowirePassTest.php

```diff
@@ -648,24 +703,33 @@ public function testInterfaceWithNoImplementationSuggestToWriteOne()
         (new AutowireRequiredMethodsPass())->process($container);
 
         $pass = new AutowirePass();
-        $pass->process($container);
+        try {
+            $pass->process($container);
+        } catch (AutowiringFailedException $e) {
+            $this->assertSame('Cannot autowire service "my_service": argument "$i" of method "Symfony\Component\DependencyInjection\Tests\Compiler\K::__construct()" references interface "Symfony\Component\DependencyInjection\Tests\Compiler\IInterface" but no such service exists. Did you create a class that implements this interface?', (string) $e->getMessage());
+            throw $e;
+        }
     }
 
     public function testProcessDoesNotTriggerDeprecations()
     {
         $this->expectException('Symfony\Component\DependencyInjection\Exception\AutowiringFailedException');
-        $this->expectExceptionMessage('Cannot autowire service "bar": argument "$foo" of method "Symfony\Component\DependencyInjection\Tests\Compiler\Bar::__construct()" references class "Symfony\Component\DependencyInjection\Tests\Compiler\Foo" but no such service exists. You should maybe alias this class to the existing "foo" service.');
         $container = new ContainerBuilder();
         $container->register('deprecated', 'Symfony\Component\DependencyInjection\Tests\Fixtures\DeprecatedClass')->setDeprecated(true);
         $container->register('foo', __NAMESPACE__.'\Foo');
         $container->register('bar', __NAMESPACE__.'\Bar')->setAutowired(true);
 
         $pass = new AutowirePass();
-        $pass->process($container);
+        try {
+            $pass->process($container);
 
-        $this->assertTrue($container->hasDefinition('deprecated'));
-        $this->assertTrue($container->hasDefinition('foo'));
-        $this->assertTrue($container->hasDefinition('bar'));
+            $this->assertTrue($container->hasDefinition('deprecated'));
+            $this->assertTrue($container->hasDefinition('foo'));
+            $this->assertTrue($container->hasDefinition('bar'));
```

Comment:
which line throws? only it should be inside the try/catch (and we should remove dead-code assertions if there are any)
same comment in similar places above

Reply:
refactored to remove `$this->expectException` which hide uncautch exceptions.

which helped me to spot dead and wrong assertions.